packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx

```diff
@@ -0,0 +1,715 @@
+  const { handleSuccess } = useRouteModal()
+
+  /**
+   * STATE
+   */
+  const { setIsOpen } = useStackedModal()
+  const [isShippingPriceEdit, setIsShippingPriceEdit] = useState(false)
+  const [customShippingAmount, setCustomShippingAmount] = useState(0)
+  const [inventoryMap, setInventoryMap] = useState<
+    Record<string, InventoryLevelDTO[]>
+  >({})
+
+  /**
+   * HOOKS
+   */
+  const { stock_locations = [] } = useStockLocations({ limit: 999 })
+  const { shipping_options = [] } = useShippingOptions({
+    limit: 999,
+    fields: "*prices,+service_zone.fulfillment_set.location.id",
+    /**
+     * TODO: this should accept filter for location_id
+     */
+  })
+
+  /**
+   * MUTATIONS
+   */
+  const { mutateAsync: confirmClaimRequest, isPending: isConfirming } = {} // useConfirmClaimRequest(claim.id, order.id)
+
+  const { mutateAsync: cancelClaimRequest, isPending: isCanceling } = {} // useCancelClaimRequest(claim.id, order.id)
+
+  const { mutateAsync: updateClaimRequest, isPending: isUpdating } = {} // useUpdateClaim(claim.id, order.id)
+
+  const {
+    mutateAsync: addInboundShipping,
+    isPending: isAddingInboundShipping,
+  } = useAddClaimInboundShipping(claim.id, order.id)
+
+  const {
+    mutateAsync: updateInboundShipping,
+    isPending: isUpdatingInboundShipping,
+  } = useUpdateClaimInboundShipping(claim.id, order.id)
+
+  const {
+    mutateAsync: deleteInboundShipping,
+    isPending: isDeletingInboundShipping,
+  } = useDeleteClaimInboundShipping(claim.id, order.id)
+
+  const { mutateAsync: addInboundItem, isPending: isAddingInboundItem } =
+    useAddClaimInboundItems(claim.id, order.id)
+
+  const { mutateAsync: updateInboundItem, isPending: isUpdatingInboundItem } =
+    useUpdateClaimInboundItem(claim.id, order.id)
+
+  const { mutateAsync: removeInboundItem, isPending: isRemovingInboundItem } =
+    useRemoveClaimInboundItem(claim.id, order.id)
+
+  const isRequestLoading =
+    isConfirming ||
+    isCanceling ||
+    isAddingInboundShipping ||
+    isUpdatingInboundShipping ||
+    isDeletingInboundShipping ||
+    isAddingInboundItem ||
+    isRemovingInboundItem ||
+    isUpdatingInboundItem ||
+    isUpdating
+
+  /**
+   * Only consider items that belong to this claim.
+   */
+  const previewItems = useMemo(
+    () =>
+      preview.items.filter(
+        (i) => !!i.actions?.find((a) => a.claim_id === claim.id)
+      ),
+    [preview.items]
+  )
+
+  const itemsMap = useMemo(
+    () => new Map(order.items.map((i) => [i.id, i])),
+    [order.items]
+  )
+
+  const previewItemsMap = useMemo(
+    () => new Map(previewItems.map((i) => [i.id, i])),
+    [previewItems]
+  )
+
+  /**
+   * FORM
+   */
+
+  const form = useForm<ReturnCreateSchemaType>({
+    defaultValues: () => {
+      const method = preview.shipping_methods.find(
+        (s) => !!s.actions?.find((a) => a.action === "SHIPPING_ADD")
+      )
+
+      return Promise.resolve({
+        inbound_items: previewItems.map((i) => ({
```

Comment:
question: Will likely need outbound items too when added right?

Reply:
yes, haven't gotten to outbound yet.